plugins/cindygl/src/js/Renderer.js

```diff
@@ -238,11 +245,21 @@ Renderer.prototype.loadTextures = function() {
         }
 }
 
-/**
- * runs shaderProgram on gl. Will render to texture in canvaswrapper 
- * or if argument canvaswrapper is not given, then to glcanvas
- */
+Renderer.prototype.functionGenerationsOk = function() {
+        for (let fname in this.generations) {
+            if (this.api.getMyfunction(fname).generation > this.generations[fname]) {
+                console.log(fname + " is outdated; forcing rebuild.");
+                return false;
+            }
+        }
+        return true;
+    }
+    /**
+     * runs shaderProgram on gl. Will render to texture in canvaswrapper
+     * or if argument canvaswrapper is not given, then to glcanvas
+     */
```

Comment:
The indentation here looks strange and confusing. Shouldn't all of this be outdented by 4 spaces? And perhaps a newline before the comment?